libyul/AsmParser.cpp

```diff
@@ -249,6 +274,8 @@ ForLoop Parser::parseForLoop()
 	forLoop.pre = parseBlock();
 	forLoop.condition = make_unique<Expression>(parseExpression());
 	forLoop.post = parseBlock();
+	bool outsideForLoopBody = exchange(m_insideForLoopBody, true);
+	auto forLoopBodyGuard = atScopeExit([=]() mutable { m_insideForLoopBody = outsideForLoopBody; });
```

Comment:
Wouldn't it be easier to just add `m_insideForLoopBody = outsideForLoopBody;` two lines below?